src/prbench/envs/tidybot/tidybot_mujoco_env.py

```
@@ -0,0 +1,609 @@
+                object_names.append(body_name)
+        object_names.sort()
+        num_objects = len(object_names)
+        print(f"Detected {num_objects} objects in scene: {object_names}")
+
+        # Shared memory for state observations
+        self.shm_state = ShmState(num_objects=num_objects, object_names=object_names)
+
+        # Shared memory for image observations
+        if self.render_images:
+            self.shm_images = []
+            for camera_id in range(model.ncam):
+                camera_name = model.camera(camera_id).name
+                width, height = model.cam_resolution[camera_id]
+                self.shm_images.append(ShmImage(camera_name, width, height))
+
+        # Start physics loop
+        mp.Process(target=self.physics_loop, daemon=True).start()
+
+        if self.render_images and self.show_images:
+            # Start visualizer loop
+            mp.Process(target=self.visualizer_loop, daemon=True).start()
+
+    def physics_loop(self):
+        """Run the physics simulation loop in a separate process."""
+        try:
+            # Create sim
+            sim = MujocoSim(
+                self.mjcf_path,
+                self.command_queue,
+                self.shm_state,
+                show_viewer=self.show_viewer,
+            )
+
+            # Start render loop
+            if self.render_images:
+                Thread(
+                    target=self.render_loop, args=(sim.model, sim.data), daemon=True
+                ).start()
+
+            # Launch sim
+            sim.launch()  # Launch in same thread as creation to avoid segfault
+        except Exception as e:
+
+            print("Physics process crashed:", e)
+            traceback.print_exc()
+
+    def render_loop(self, model, data):
+        """Run the rendering loop for camera images in a separate process."""
+        # Set up renderers
+        renderers = [Renderer(model, data, shm_image) for shm_image in self.shm_images]
```

this was silently failing before and it was difficult to track down, so add a try / except like above